### PR TITLE
Replace SubmitGem implementation with JRuby based one

### PR DESCRIPTION
**Pull request by Sammers21**
As a proof of concept for #13

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#14](https://github.com/artipie/gem-adapter/issues/14) is assigned to @g4s8[/z](https://www.0crat.com/u/g4s8), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/af1539f6-70a4-42b4-997b-da094cbc1280); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

File: src/main/java/com/artipie/gem/GemSlice.java [hunk 1]
```diff
@@ -70,4 +88,30 @@ public GemSlice(final Storage storage) {
...
         );
     }
+
+    /**
+     * Lookup an instance of slice, implemented with JRuby.
+     * @param rclass The name of a slice class, implemented in JRuby.
+     * @param runtime The JRuby runtime.
+     * @return The Slice.
+     */
+    private static Slice rubyLookUp(final String rclass, final Ruby runtime) {
```
Comment (g4s8) [inline src/main/java/com/artipie/gem/GemSlice.java:98]: @Sammers21 it seems to be slow operation, I think we need to lazy initialize this ruby slice to create it on first call. If we create it in the constructor, the server will load it even if gem repository won't be enabled in configuration.

File: src/main/java/com/artipie/gem/GemSlice.java [hunk 2]
```diff
@@ -70,4 +88,30 @@ public GemSlice(final Storage storage) {
...
+        try {
+            final RubyRuntimeAdapter evaler = JavaEmbedUtils.newRuntimeAdapter();
+            final String script = IOUtils.toString(
+                GemSlice.class.getResourceAsStream(String.format("/%s.rb", rclass)),
+                StandardCharsets.UTF_8
+            );
+            evaler.eval(runtime, script);
+            Object rfj = evaler.eval(runtime, String.format("%s.new()", rclass));
```
Comment (g4s8) [inline src/main/java/com/artipie/gem/GemSlice.java:106]: @Sammers21 what `rfj` means?

File: src/main/java/com/artipie/gem/GemSlice.java [hunk 3]
```diff
@@ -70,4 +88,30 @@ public GemSlice(final Storage storage) {
...
+            rfj = JavaEmbedUtils.rubyToJava(
```
Comment (g4s8) [inline src/main/java/com/artipie/gem/GemSlice.java:107]: @Sammers21 it's not a good practice to reuse variable for two different objects

File: src/main/java/com/artipie/gem/GemSlice.java (hunk 2 above)
Comment (Sammers21) [reply src/main/java/com/artipie/gem/GemSlice.java:106]: Ruby from Java

File: src/main/java/com/artipie/gem/GemSlice.java (hunk 3 above)
Comment (Sammers21) [reply src/main/java/com/artipie/gem/GemSlice.java]: @g4s8 , fixed

File: src/main/java/com/artipie/gem/GemSlice.java (hunk 2 above)
Comment (Sammers21) [reply src/main/java/com/artipie/gem/GemSlice.java]: The variable has been removed

File: src/main/java/com/artipie/gem/GemSlice.java (hunk 1 above)
Comment (Sammers21) [reply src/main/java/com/artipie/gem/GemSlice.java:101]: @g4s8 , a puzzle for that has been created.

**Comment by 0crat** [comment]
Job was finished in 0 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
Job audit: `ARC` didn't merge or close PR, performer @g4s8[/z](https://www.0crat.com/u/g4s8)[/z](https://www.0crat.com/u/g4s8) didn't make CR comments

**Pull request merged**